p2p/host/pstoremanager/pstoremanager_test.go

```diff
@@ -86,11 +86,18 @@ func TestClose(t *testing.T) {
 
 	emitter, err := eventBus.Emitter(new(event.EvtPeerConnectednessChanged))
 	require.NoError(t, err)
+
+	sub, err := eventBus.Subscribe(&event.EvtPeerConnectednessChanged{})
+	require.NoError(t, err)
+
 	require.NoError(t, emitter.Emit(event.EvtPeerConnectednessChanged{
 		Peer:          "foobar",
 		Connectedness: network.NotConnected,
 	}))
-	time.Sleep(10 * time.Millisecond) // make sure the event is sent before we close
+
+	// make sure the event is sent before we close
+	<-sub.Out()
```

Comment:
Should we have a select with a timeout here?

Reply:
Also don’t feel strongly. I generally prefer it because the error message is nicer.